autocertkit/utils.py

```diff
@@ -1932,6 +1932,29 @@ def xml_to_dicts(xml, tag):
     log.debug(result)
     return result
 
+def json_loads(json_data):
+    def process_dict_keys(d):
+        new_d = {}
+        for key in d.iterkeys():
+            new_key = str(key.replace(" ","_"))
+            new_d[new_key] = d[key]
+        return new_d
+
+    def process_values(item):
+        if isinstance(item, unicode):
+            item = str(item)
+        elif isinstance(item, list):
+            for elem in item:
+                elem = process_values(elem)
+        elif isinstance(item, dict):
+            item = process_dict_keys(item)
+            for key in item.iterkeys():
+                item[key] = process_values(item[key])
+        return item
+
+    data = json.loads(json_data, object_hook=process_values) if json_data else []
+    return [data] if isinstance(data, dict) else data
+
 def call_ack_plugin(session, method, args={}, host=None):
```

Comment:
I believe we can parse json into proper python object inside call_ack_plugin.
Main reason we delegated xml parsing into caller is that we do not know whether this is a simple string, list or dict. Now we do not need to care as json supports all of them.